src/orchard/namespace.clj

```diff
@@ -148,9 +151,27 @@
             io/resource ;; can return nil for Emacs backup files, for example
             read-ns-name)))
 
+(defn- find-cljc-files-efficiently
+  "Finds .clj/c files as efficiently as possible.
+
+  In particular, avoids visiting and operating on files,
+  as early as possible."
+  [^File dir]
+  (let [start-path (.toPath dir)
```

Comment:
I'd suggest to document in more details the implementation details here, as I'm not familiar with neither `BiPredicate` and the related Java APIs, so it's hard for me to comment on the quality of the implementation. But I'd definitely like to understand better why are we using those.

Reply:
I (hopefully) improved the docstring.

The improvement gains are obtained as follows:

* We have an optimized 2-arg predicate (BiPredicate) that shortcircuits / avoids garbage creation
* We return fewer files (only .clj/c files)
* Because it receives fewer files, the upstream consumer (`orchard.java.classpath/classpath-seq`) doesn't uselessly create more computations/garbage e.g.:

https://github.com/clojure-emacs/orchard/blob/951fdbe26cf5146ef070b905a99508e6ef9997e1/src/orchard/java/classpath.clj#L91-L93

For the affected user's resource dir, we now run the code above for 50 .clj/c files instead of 1M irrelevant files.

Other than that, there's no particular magic in the Java interop, it's a fairly vanilla way of listing files (vs. `file-seq` which lacks any filtering options).

If we wanted to further optimize things, we could look into performing a Unix `ls | grep` with C interop, which I believe would be faster since it avoids creating 1M `Path`/`File`objects.

But things seem usable as-is at the moment.

Reply:
Yeah, the interop code is not particularly complex - I was mostly wondering if we need it at all and can get similar results with Clojure's file APIs.

Reply:
The docstring is fine in its current form.